src/main/java/org/folio/service/FundsDistributionService.java

```diff
@@ -29,54 +30,71 @@ public class FundsDistributionService {
 
-        final EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
-          CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
-          MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
+        EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
+        CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
+        MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
 
-          if (FundDistribution.DistributionType.PERCENTAGE.equals(holder.getFundDistribution().getDistributionType()) && !remainder.isZero()) {
-            initialAmount = initialAmount.add(smallestUnit);
-            remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
-          }
+        if (holder.getFundDistribution().getDistributionType() == FundDistribution.DistributionType.PERCENTAGE && !remainder.isZero()) {
+          initialAmount = initialAmount.add(smallestUnit);
+          remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
+        }
 
-          MonetaryAmount expended = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountExpended).map(aDouble -> Money.of(aDouble, fyCurrency))
-              .orElse(Money.zero(fyCurrency));
-          MonetaryAmount awaitingPayment = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountAwaitingPayment)
-              .map(aDouble -> Money.of(aDouble, fyCurrency)).orElse(Money.zero(fyCurrency));
-          MonetaryAmount amount = calculateEncumbranceEffectiveAmount(initialAmount, expended, awaitingPayment, fyCurrency);
+        MonetaryAmount expended = extractAmount(holder, fyCurrency, Encumbrance::getAmountExpended);
+        MonetaryAmount credited = extractAmount(holder, fyCurrency, Encumbrance::getAmountCredited);
+        MonetaryAmount awaitingPayment = extractAmount(holder, fyCurrency, Encumbrance::getAmountAwaitingPayment);
 
-          holder.getNewEncumbrance().setAmount(amount.getNumber().doubleValue());
-          holder.getNewEncumbrance().getEncumbrance().setInitialAmountEncumbered(initialAmount.getNumber().doubleValue());
+        MonetaryAmount amount = FinanceUtils.calculateEncumbranceEffectiveAmount(
+          initialAmount,
+          expended,
+          credited,
+          awaitingPayment,
+          fyCurrency
+        );
+
+        holder.getNewEncumbrance().setAmount(amount.getNumber().doubleValue());
+        holder.getNewEncumbrance().getEncumbrance().setInitialAmountEncumbered(initialAmount.getNumber().doubleValue());
 
       }
     });
     return holders;
   }
 
+  public static MonetaryAmount extractAmount(EncumbranceRelationsHolder holder,
```

Comment:
this method is used only in this class from non-static context, so we can make it as non static

Reply:
done